lib/ansible/modules/cloud/cloudscale/cloudscale_server_group.py

```diff
@@ -0,0 +1,203 @@
+---
+href:
+  description: API URL to get details about this server group
+  returned: if available
+  type: str
+  sample: https://api.cloudscale.ch/v1/server-group/cfde831a-4e87-4a75-960f-89b0148aa2cc
+uuid:
+  description: The unique identifier for this server
+  returned: always
+  type: str
+  sample: cfde831a-4e87-4a75-960f-89b0148aa2cc
+name:
+  description: The display name of the server group
+  returned: always
+  type: str
+  sample: load balancers
+type:
+  description: The type the server group
+  returned: if available
+  type: str
+  sample: anti-affinity
+servers:
+  description: A list of servers that are part of the server group.
+  returned: if available
+  type: list
+  sample: []
+state:
+  description: State of the server group.
+  returned: always
+  type: str
+  sample: present
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.cloudscale import AnsibleCloudscaleBase, cloudscale_argument_spec
+
+
+class AnsibleCloudscaleServerGroup(AnsibleCloudscaleBase):
+
+    def __init__(self, module, namespace):
+        super(AnsibleCloudscaleServerGroup, self).__init__(module)
+        self._info = {}
+
+    def _init_container(self):
+        return {
+            'uuid': self._module.params.get('uuid') or self._info.get('uuid'),
+            'name': self._module.params.get('name') or self._info.get('name'),
+            'state': 'absent',
+        }
+
+    def _create_server_group(self, server_group):
+        self._result['changed'] = True
+        data = {
+            'name': self._module.params.get('name'),
+            'type': self._module.params.get('type'),
+        }
+        if not self._module.check_mode:
+            server_group = self._post('server-groups', data)
+        return server_group
+
+    def _update_server_group(self, server_group):
+        data = {
+            'name': self._module.params.get('name'),
+        }
+        if server_group['name'] != data['name']:
+            self._result['changed'] = True
+
+            if not self._module.check_mode:
+                self._patch('server-groups/%s' % server_group['uuid'], data)
+                server_group = self.get_server_group()
+        return server_group
+
+    def get_server_group(self):
+        self._info = self._init_container()
+
+        uuid = self._info.get('uuid')
+        if uuid is not None:
+            server_group = self._get('server-groups/%s' % uuid)
+            if server_group:
+                self._info.update(server_group)
+                self._info.update(dict(state='present'))
+
+        else:
+            name = self._info.get('name')
+            if name is not None:
+                server_groups = self._get('server-groups') or []
+                for server_group in server_groups:
+                    if server_group['name'] == name:
+                        self._info.update(server_group)
+                        self._info.update(dict(state='present'))
+                        break
```

Comment:
The API currently allows multiple groups with the same name. Do we need some logic similar to the logic in `cloudscale_server` to detect this? Otherwise this could lead to surprising results...
And tests for this case would be nice too.